firewood/src/merkle/node.rs

```diff
@@ -375,8 +384,27 @@ impl Storable for Node {
             None
         };
 
+        let mut start_index = TRIE_HASH_LEN + 1;
```

Comment:
We should be using a cursor here instead (that's if you don't want to use `bytemuck`)